File: api/book.py
from flask import Flask, jsonify, request
import requests
from bs4 import BeautifulSoup

app = Flask(__name__)

@app.route('/book', methods=['GET'])
def get_book_data():
    # Get the ISBN from the query parameters
    isbn = request.args.get('isbn')
    if not isbn:
        return jsonify({'message': 'ISBN is required'}), 400

    try:
        # Define the URL for the textbook website
        url = f"https://www.textbookcenter.com/search/?q={isbn}"
        
        # Fetch the page content
        response = requests.get(url)
        if response.status_code != 200:
            return jsonify({'message': 'Failed to fetch data from external source'}), 500
        
        # Parse the HTML content using BeautifulSoup
        soup = BeautifulSoup(response.text, 'html.parser')
        title = soup.select_one('.product-title')
        authors = soup.select_one('.product-author')

        if not title or not authors:
            return jsonify({'message': 'Book not found'}), 404

        # Extract the text and clean it
        title_text = title.text.strip()
        authors_text = authors.text.strip()

        # Return the book data as JSON
        return jsonify({
            'title': title_text,
            'authors': [a.strip() for a in authors_text.split(',')],
            'source': 'Text Book Center'
        })

    except Exception as e:
        print(f"Error: {e}")
        return jsonify({'message': 'Internal server error'}), 500

if __name__ == '__main__':
    app.run(debug=True)

File: api/book.js
const axios = require("axios");
const cheerio = require("cheerio");

module.exports = async (req, res) => {
  const { isbn } = req.query;  // Extract the ISBN from the query string

  // Ensure ISBN is provided
  if (!isbn) {
    return res.status(400).json({ message: "ISBN is required" });
  }

  const fetchTextBookCenter = async () => {
    try {
      const response = await axios.get(`https://www.textbookcenter.com/search/?q=${isbn}`);
      const $ = cheerio.load(response.data);
      const title = $('.product-title').first().text().trim();
      const authors = $('.product-author').first().text().trim();

      if (title && authors) {
        return {
          title,
          authors: authors.split(',').map(a => a.trim()),
          source: 'Text Book Center',
        };
      } else {
        console.error("No title or authors found for ISBN:", isbn);
        return null;
      }
    } catch (err) {
      console.error("Error fetching data from Textbook Center:", err.message);
      return null;
    }
  };

  try {
    const result = await fetchTextBookCenter();

    if (result) {
      res.status(200).json(result);  // Send back the book data if found
    } else {
      res.status(400).json({ message: "Book not found" });  // Send error if not found
    }
  } catch (err) {
    console.error("Error handling the request:", err.message);
    res.status(500).json({ message: "A server error occurred" });
  }
};


